x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/retry/RetryException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http.retry;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.ElasticsearchWrapperException;
+
+public class RetryException extends ElasticsearchException implements ElasticsearchWrapperException {
```

Comment:
> but by being an ElasticsearchException it does gain some of those inherent features

Sounds good I'll leave it extending `ElasticsearchException` 👍 

> Can RetryException be package private? Is it used outside of this package?

Yeah it could at the moment but pretty soon I think we'll want to define more logic around when we retry. My guess is that'll end up in the various `xpack.inference.external.action.*` (e.g. openai/huggingface etc) packages but maybe the retry logic would be the same for all services and we could just move it to `xpack.inference.external.http.retry` where `RetryException` already lives.

I'll add a TODO item but how about for now we wait until we define that specific logic and move it then if it still not needed outside of the package.